### ISSUE TITLE
Adding the CReLU Activation

### ISSUE THREAD

**Issue by keskarnitish** [pytorch/pytorch#1327]
I was trying to reproduce some of the experiments of [the new paper on shattered gradients](https://arxiv.org/abs/1702.08591) and needed to use the [CReLU activation](https://arxiv.org/abs/1603.05201) which was unavailable in PyTorch. Unfortunately, it doesn't seem to be a trivial extension of ReLU to code. (Should?) Could we add this functionality to PyTorch?

Implementations of CReLU can be found in:
[Lua Torch](https://github.com/torch/nn/blob/master/CReLU.lua)
[Chainer](http://docs.chainer.org/en/latest/_modules/chainer/functions/activation/crelu.html)
[Tensorflow](https://www.tensorflow.org/api_docs/python/tf/nn/crelu)

If I am not mistaken, this function would be added to [``nn/_functions/thnn/activation``](https://github.com/pytorch/pytorch/blob/master/torch/nn/_functions/thnn/activation.py)?

### Update:
For CNNs and Linear layers, I think I have a temporary solution.

```
class CReLU(nn.Module):
    def __init__(self):
        super(CReLU, self).__init__()
    def forward(self, x):
        return torch.cat((F.relu(x), F.relu(-x)), 1)
```

Obviously, there are some issues with this (hard-coded `axis` argument, doesn't allow inplace operations etc.)

**Comment by apaszke** [comment]
As you found, it's unnecessary to make this a function, because it can be easily implemented using existing autograd ops.

**Comment by keskarnitish** [comment]
Sounds good to me! We can close this issue if you'd like.

**Comment by progirep** [comment]
Ok, so it can be easily be implemented using existing autograd operations. But how exactly (without avoiding the draw-backs that @keskarnitish mentioned)? LuaTorch has such a function, so perhaps it makes sense to add it to pytorch as well?

**Comment by surgan12** [comment]
Is adding this still relevant ?

**Comment by progirep** [comment]
> Is adding this still relevant ?

Well, to me it is. Alternatively, a documented implementation using Autograd that
avoids the drawbacks that keskarnitish mentioned would also do the trick.

**Comment by MohamedAliRashad** [comment]
I agree with @progirep and  think this issue needs to be reopened and implemented ..... PyTorch is nothing less than Tensorflow or others in implementing such functions.

**Comment by mueller91** [comment]
also would appreciate a **crelu**
